src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2573,8 +2582,12 @@ internal void DoCommand(EventCommandEventArgs commandArgs)
             }
 
             // PRECONDITION: We should be holding the EventListener.EventListenersLock
+            Debug.Assert(Monitor.IsEntered(EventListener.EventListenersLock));
             // We defer commands until we are completely inited.  This allows error messages to be sent.
-            Debug.Assert(m_completelyInited);
+            // Technically, by this point initialization should be complete, but m_completelyInited
```

Comment:
Similar to above, I don't think you need to make any claims about initialization being 'complete', just that m_etwProvider has been setup. That allows us to report errors if they occur.

Reply:
Done!